packages/utils-general/src/blockchain-facade/AssetTypeService.ts

```diff
@@ -88,7 +88,48 @@ export class IRECAssetService implements IAssetService {
     }
 
     includesAssetType(current: string, requested: string[]): boolean {
-        return requested.some(requestedAssetType => current === requestedAssetType);
+        console.log('requested', requested);
+        const decodedRequested = requested.map(type => [...this.decode([type])[0]]);
+
+        console.log({
+            current,
+            decodedRequested
+        });
+
+        const requestedFlattened = decodedRequested.filter(type => {
+            if (type.length === 2 || type.length === 3) {
+                // change to 3
+                return true;
+            }
+
+            return !decodedRequested.some(
+                nestedType =>
+                    nestedType[0] === type[0] &&
+                    type.length === 1 &&
+                    nestedType.length !== type.length
+            );
+        });
+
+        console.log({
+            requestedFlattened
+        });
+
+        for (const assetType of requestedFlattened) {
+            const decodedTypeToCheck = [...this.decode([current])[0]];
+
+            if (
+                JSON.stringify(assetType) === JSON.stringify(decodedTypeToCheck) ||
+                (assetType.length < decodedTypeToCheck.length &&
+                    assetType.every(
+                        (type, index) =>
+                            type === decodedTypeToCheck[index] || !decodedTypeToCheck[index]
+                    ))
+            ) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
This looks suspiciously complex, why do you need that? Please add the unit test with input that requires those changes.

Reply:
Yep, it's in progress. I need this because:

If only wind (level 1) is selected it should match with wind;onshore (level 2)

same with level 3 being matched with level 2, unless level 3 specificity is present for that concrete category

Reply:
Yes, this is by design. This method is used by matcher to check for compatible asset types. Now if the demand is broad and requests `Wind` then whatever level 2 or level 3 type of `Wind` is defined by `ProducingAsset`, it matches.

Take a look at tests here: https://github.com/energywebfoundation/origin/blob/5a93304f837c317f6940f35ee37113fef66616eb/packages/utils-general/src/test/AssetTypeService.test.ts#L71

Reply:
So instead of changing the impl of this method please add the new one for the purpose.

Reply:
The test is there but it was failing, you can check on master. Also I believe, it doesn't check correctly when `Wind` and `Wind;Onshore` for Wind:Offshore would be in the same array, because it would match `Wind;Offshore` with `Wind`, while it shouldn't match, because there's a level 2 specificity passed - `Wind;Onshore`.

Reply:
Yes, you are right, test was failing. #90 fixes it.